src/components/ViewerComponent.vue

```diff
@@ -67,19 +80,86 @@ export default {
 			type: Boolean,
 			default: false,
 		},
+		permissions: {
+			type: String,
+			default: '',
+		},
+		source: {
+			type: String,
+			default: undefined,
+		},
+	},
+	data() {
+		return {
+			content: '',
+		}
+	},
+	computed: {
+		/** @return {boolean} */
+		isEditable() {
+			return this.permissions.includes('W')
+		},
+
+		/** @return {boolean} */
+		readerComponent() {
+			return this.mime === 'text/markdown' ? RichTextReader : PlainTextReader
+		},
+	},
+
+	watch: {
+		source() {
+			this.loadFileContent()
+		},
+		active() {
+			this.loadFileContent()
+		},
```

Comment:
I don't quite understand why this is needed. The `content` should already be fetched during the `mounted` handler. Why do we need to fetch it again if `active` is toggled?
Do we also need to fetch it if `active` turns false?

I assume this won't matter anymore if the `files` list in the viewer is empty as i suspect active is toggled when a `previous` file becomes `current`.

Reply:
Yes, this seemed a leftover from the initial attempt. Removed now